FreeRTOS/FreeRTOS-Plus-TCP
# dev/IPv6: DHCPv6 repaired and tested

Description
-----------
The DHCPv6 module was developed but rarely used and tested in real hardware.

It is now tested with a protocol tester, and also we have simulated a situation in which packets get lost.

Also added the possibility to store multiple DNS entries.

Earlier initialisation of IP-address of endpoints: in FreeRTOS_Routing.c, the IP-address will be initialised in case DHCP or RA are not used:
~~~
/* IPv4 */
if( pxEndPoint->bits.bWantDHCP == 0U )
{
    pxEndPoint->ipv4_settings.ulIPAddress = ulIPAddress;
}

/* IPv6 */
if( ( pxEndPoint->bits.bWantDHCP == 0 ) && ( pxEndPoint->bits.bWantRA == 0 ) )
{
    /* Using a static IP-address. */
    memcpy( pxEndPoint->ipv6_settings.xIPAddress.ucBytes, pxIPAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
}
~~~
The above is necessary because DHCPv6 wants to use a local-link IP-adddress.

Test Steps
-----------
Start a DHCPv6 server and activate DHCP:
~~~
/* In FreeRTOSIPConfig.h */
#define ipconfigUSE_DHCP    1
#define ipconfigUSE_DHCPv6  1
/* In main.c: */
pxEndPoint->bits.bWantDHCP = pdTRUE;
~~~


Checklist:
----------




By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
